### PR TITLE
Add recipe for unbranded firefox on linux

### PR DESCRIPTION
**Pull request by birdsarah**
This PR supercedes #10100

It incorporates @isuruf's suggestions from that PR.

### CODE REVIEW

File: recipes/firefox-unbranded/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  binary_relocation:
+    - bin/firefox
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+
+test:
+  # Unfortunately can't test on linux due to conda-forge build infrastructure 
+  # not having gtk3. https://github.com/conda-forge/staged-recipes/issues/9314
```
Comment (isuruf) [inline recipes/firefox-unbranded/meta.yaml:27]: We have `gtk3` now.

Comment (birdsarah) [reply recipes/firefox-unbranded/meta.yaml]: If we can solve it https://github.com/conda-forge/firefox-feedstock/pull/18 I'll fix it here too.

**Comment by birdsarah** [comment]
OK. Well that's at least enlightening! The last commit makes this PR identical to the current [firefox-feedstock](https://github.com/conda-forge/firefox-feedstock). But it fails. The firefox-feedstock is happily being used by me and others.

@isuruf @scopatz @msarahan @ocefpaf - just pinging friends in case anyone has any thoughts.

**Comment by birdsarah** [comment]
i'm going to try again, with latest improvements from @bollwyvl on firefox-feedstock repo.

**Pull request closed without merging**